bundles/net.tourbook/src/net/tourbook/ui/UI.java

```diff
@@ -547,6 +566,51 @@ private static void disableAllControlsInternal(final Composite container) {
       }
    }
 
+   private static Map<Long, String> fetchTourTagsAccumulatedValues() {
+
+      final String sqlQuery = "SELECT" + NEW_LINE //                                                                        //$NON-NLS-1$
+            + "jTdataTtag.TOURTAG_TAGID," + NEW_LINE // //$NON-NLS-1$
+            + "SUM(tourData.TOURDISTANCE) AS TOTALDISTANCE," + NEW_LINE // //$NON-NLS-1$
+            + "SUM(tourData.TOURDEVICETIME_RECORDED) AS TOTALRECORDEDTIME" + NEW_LINE //                                                                   //$NON-NLS-1$
+            + "FROM " + TourDatabase.JOINTABLE__TOURDATA__TOURTAG + " jTdataTtag" + NEW_LINE //                                                                        //$NON-NLS-1$ //$NON-NLS-2$
+            + "INNER JOIN " + TourDatabase.TABLE_TOUR_DATA + NEW_LINE //                                                       //$NON-NLS-1$
+            + "ON jTdataTtag.TOURDATA_TOURID = tourData.TOURID" + NEW_LINE //                                                                       //$NON-NLS-1$
+            + "GROUP BY jTdataTtag.TOURTAG_TAGID"; //$NON-NLS-1$
+
+      final Map<Long, String> tourTagsAccumulatedValues = new HashMap<>();
+
+      try (Connection connection = TourDatabase.getInstance().getConnection();
+            final PreparedStatement preparedStatement = connection.prepareStatement(sqlQuery)) {
+
+         final ResultSet result = preparedStatement.executeQuery();
+
+         while (result.next()) {
+
+            final long tourTagId = result.getLong(1);
+            float usedMiles = result.getInt(2);
+            final long usedHours = result.getInt(3);
+
+            final StringBuilder tagAccumulatedValues = new StringBuilder();
+
+            tagAccumulatedValues.append(Math.round(usedHours / 3600f));
+            tagAccumulatedValues.append(Messages.tour_editor_label_time_unit);
+
+            tagAccumulatedValues.append(UI.NEW_LINE);
+
+            usedMiles = usedMiles / 1000 / net.tourbook.common.UI.UNIT_VALUE_DISTANCE;
+            tagAccumulatedValues.append(Math.round(usedMiles));
+            tagAccumulatedValues.append(net.tourbook.common.UI.UNIT_LABEL_DISTANCE);
+
+            tourTagsAccumulatedValues.put(tourTagId, tagAccumulatedValues.toString());
+         }
+
+      } catch (final SQLException e) {
+
+         SQL.showException(e, sqlQuery);
+      }
+      return tourTagsAccumulatedValues;
+   }
+
```

Comment:
I think the `TagManager` would be a better place for this tag code, there is already other SQL code